test_cases/stdlib/check_concurrent_futures.py

```diff
@@ -0,0 +1,23 @@
+from __future__ import annotations
+
+from collections.abc import Iterator
+from concurrent.futures import Future, ThreadPoolExecutor, as_completed
+from typing_extensions import assert_type
+
+
+class Parent:
+    ...
+
+
+class Child(Parent):
+    ...
+
+
+def check_as_completed_covariance() -> None:
```

Comment:
As far as I can tell, the original PR that just makes `Future` covariant would pass this test. The test basically checks that `Future` acts covariantly, and that's all it does.

Can you add a test that the previous PR fails and this PR passes? If you can't, or if the test becomes ends up so "niche" it doesn't do anything practical, I think it's a clear sign that we should just make the thing covariant. My instinct is that it's probably what we should do anyway, because it is practical and simple, but a good test case could convince me otherwise.